File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
N, M = map(int, input().split())
bagu = list(range(1, N + 1))
for _ in range(M):
    i, j = map(int, input().split())
    i = i - 1 # bagu의 1은 리스트 0번이라서 i-1
    basket = bagu[i:j] # 새로운 변수에 슬라이싱 할당
    basket.reverse() # 거꾸로
    bagu[i:j] = basket # 거꾸로한거 다시 원래슬라이싱에 할당
        
print(*bagu)

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
const fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, M] = input[0].split(" ");
let basket = Array.from({ length: N }, (_, index) => index + 1);

for (let i = 1; i <= M; i++) {
  const [s, e] = input[i].split(" ");

  let ref = basket.slice(s - 1, e);
  ref.reverse();
  basket.splice(s - 1, e - s + 1, ...ref);
}

console.log(basket.join(" "));
